java/ingest/src/test/java/sleeper/ingest/testutils/IngestRecordsTestDataHelper.java

```diff
@@ -0,0 +1,295 @@
+            records.add(record1);
+            Record record2 = new Record();
+            record2.put("key", 5L);
+            record2.put("value1", 4L);
+            record2.put("value2", 6L);
+            records.add(record2);
+        }
+        return records;
+    }
+
+    public static List<Record> getRecordsForAggregationIteratorTest() {
+        List<Record> records = new ArrayList<>();
+        Record record1 = new Record();
+        record1.put("key", new byte[]{1, 1});
+        record1.put("sort", 2L);
+        record1.put("value", 1L);
+        Record record2 = new Record();
+        record2.put("key", new byte[]{11, 2});
+        record2.put("sort", 1L);
+        record2.put("value", 1L);
+        Record record3 = new Record();
+        record3.put("key", new byte[]{1, 1});
+        record3.put("sort", 2L);
+        record3.put("value", 6L);
+        Record record4 = new Record();
+        record4.put("key", new byte[]{11, 2});
+        record4.put("sort", 1L);
+        record4.put("value", 3L);
+        records.add(record1);
+        records.add(record2);
+        records.add(record3);
+        records.add(record4);
+        return records;
+    }
+
+    public static List<Record> getRecordsOscillatingBetween2Partitions() {
+        List<Record> records = new ArrayList<>();
+        Record record1 = new Record();
+        record1.put("key1", 0);
+        record1.put("key2", 1L);
+        record1.put("value1", 2L);
+        record1.put("value2", 1L);
+        Record record2 = new Record();
+        record2.put("key1", 0);
+        record2.put("key2", 20L);
+        record2.put("value1", 200L);
+        record2.put("value2", 100L);
+        Record record3 = new Record();
+        record3.put("key1", 100);
+        record3.put("key2", 1L);
+        record3.put("value1", 20000L);
+        record3.put("value2", 10000L);
+        Record record4 = new Record();
+        record4.put("key1", 100);
+        record4.put("key2", 50L);
+        record4.put("value1", 2000000L);
+        record4.put("value2", 1000000L);
+        records.add(record1);
+        records.add(record2);
+        records.add(record3);
+        records.add(record4);
+        return records;
+    }
+
+    public static List<Record> readRecordsFromParquetFile(String filename, Schema schema) throws IOException {
+        ParquetReader<Record> reader = new ParquetRecordReader.Builder(new Path(filename), schema).build();
+        List<Record> readRecords = new ArrayList<>();
+        CloneRecord cloneRecord = new CloneRecord(schema);
+        Record record = reader.read();
+        while (null != record) {
+            readRecords.add(cloneRecord.clone(record));
+            record = reader.read();
+        }
+        reader.close();
+        return readRecords;
+    }
+
+    public static <T> void assertSketches(Stream<T> valueStream, Schema schema, String keyToCompare, ItemsSketch<T> expectedSketch, FileInfo fileInfo) throws IOException {
```

Comment:
This function is a bit odd because it calculates a sketch, and asserts that the result is the same as the one it calculated. Rather than asserting the result, it's asserting that it does the same thing that you get from ItemsSketch.update.

Rather than building an expected sketch, we could assert the values for the min, max and quantiles explicitly?

We could also clean up the function signature so that there are fewer parameters and a more descriptive name, but asserting the values explicitly might help with that.

I still think it's a good improvement to extract this function out as it is.

Reply:
I've refactored this in the latest commit to explicitly compare min/max/quantile values in the sketch